src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/service/TelemetryService.kt

```diff
@@ -19,17 +19,30 @@ class TelemetryService(
   }
 
   fun trackGoalCreateEvent(goal: Goal) {
-    telemetryClient.trackEvent(GOAL_CREATE_EVENT, goal.goalCreateEventCustomDimensions())
-    with(Span.current()) {
-      goal.goalCreateEventCustomDimensions().forEach { (key, value) -> setAttribute(key, value) }
-      addEvent("$GOAL_CREATE_EVENT-via-span")
-    }
+    telemetryClient.trackEvent(
+      GOAL_CREATE_EVENT,
+      getExistingCustomDimensions().plus(createEventCustomDimensions(goal)),
+    )
   }
 
-  private fun Goal.goalCreateEventCustomDimensions(): Map<String, String> =
-    mapOf(
-      "status" to this.status.name,
-      "stepCount" to this.steps.size.toString(),
-      "reference" to this.reference.toString(),
-    )
+  /**
+   * Returns a map of data representing the custom dimensions for the `goal-create` event.
+   */
+  private fun createEventCustomDimensions(goal: Goal): Map<String, String> =
+    with(goal) {
+      mapOf(
+        "status" to status.name,
+        "stepCount" to steps.size.toString(),
+        "reference" to reference.toString(),
+      )
+    }
+
+  /**
+   * Returns a map of data representing any custom dimensions already set via the opentelemetry API.
+   */
+  private fun getExistingCustomDimensions(): Map<String, String> =
+    Baggage.current().asMap().mapValues {
```

Comment:
I dont like the opentelemetry API .... Span, Baggage ... what stupid classnames!